database/models/user.model.ts

```diff
@@ -81,30 +89,36 @@ const UserSchema = new Schema(
       type: String,
       required: false,
     },
-    learningGoals: {
-      type: [String],
-      required: false,
-    },
+    learningGoals: [
+      {
+        type: GoalSchema,
+        required: false,
+      },
+    ],
     role: {
       type: String,
       required: false,
       default: "user",
     },
-    technologyStack: {
-      type: technologyStackSchema,
+    technologies: {
+      type: [String],
       required: false,
     },
-    experienceLevel: {
+
+    experiences: {
       type: [String],
       required: false,
     },
-    availability: {
-      type: [Date],
-      required: false,
+    startTime: {
+      type: Date,
+      required: true,
     },
-    tags: {
-      type: [Schema.Types.ObjectId],
-      ref: "Tag",
+    endTime: {
+      type: Date,
+      required: true,
+    },
```

Comment:
![image](https://github.com/agordon123/git_note/assets/4108484/efa31b16-4fb4-4bec-9240-a1ee8df0fb8f)

Start and endtime are now required. But when you register a user you don't supply these values. Either make them optional or supply a date as the default otherwise your application is broken.